codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ValueEntitySourceGenerator.scala

```diff
@@ -23,6 +23,78 @@ import com.lightbend.akkasls.codegen.Format
 object ValueEntitySourceGenerator {
   import com.lightbend.akkasls.codegen.SourceGeneratorUtils._
 
+  /**
+   * Generate Scala sources the user view source file.
+   */
+  def generateUnmanaged(valueEntity: ModelBuilder.ValueEntity, service: ModelBuilder.EntityService): Iterable[File] = {
+    val generatedSources = Seq.newBuilder[File]
+
+    val packageName = service.fqn.parent.scalaPackage
+    val packagePath = packageAsPath(packageName)
+
+    //FIXME add ValueEntity implementation
+
+    generatedSources.result()
+  }
+
+  /**
+   * Generate Scala sources for provider, handler, abstract baseclass for a view.
+   */
+  def generateManaged(valueEntity: ModelBuilder.ValueEntity, service: ModelBuilder.EntityService): Iterable[File] = {
+    val generatedSources = Seq.newBuilder[File]
+
+    val packageName = valueEntity.fqn.parent.scalaPackage
+    val packagePath = packageAsPath(packageName)
+
+    generatedSources += File(
+      s"$packagePath/${valueEntity.abstractEntityName}.scala",
+      abstractEntity(valueEntity, service))
+    //FIXME add Handler and Provider
+
+    generatedSources.result()
+  }
+
+  private[codegen] def abstractEntity(
+      valueEntity: ModelBuilder.ValueEntity,
+      service: ModelBuilder.EntityService): String = {
+    val stateType = valueEntity.state.fqn.name
+    val packageName = valueEntity.fqn.parent.scalaPackage
+    val className = valueEntity.fqn.name
+
+    val imports = generateCommandImports(
+      service.commands,
+      valueEntity.state,
+      packageName,
+      otherImports = Seq(
+        "com.akkaserverless.scalasdk.valueentity.ValueEntity",
+        "com.akkaserverless.scalasdk.valueentity.ValueEntity.Effect"),
+      false)
+
+    val methods = service.commands
+      .map { cmd =>
+        val methodName = cmd.name
+
+        val inputType = cmd.inputType.name
+        val outputType = cmd.outputType.name
+
+        s"""|/** Command handler for "${cmd.name}". */
+            |def ${lowerFirst(methodName)}(currentState: $stateType, ${lowerFirst(cmd.inputType.name)}: $inputType): Effect[$outputType]
+            |""".stripMargin
+
+      }
+
+    s"""|package $packageName
+        |
+        |$imports
+        |
+        |/** A value entity. */
+        |abstract class Abstract$className extends ValueEntity[$stateType] {
```

Comment:
```suggestion
        |abstract class ${className} extends ValueEntity[${typeName(stateType)}] {
```

Reply:
here it should be `abstractEntityName`

Reply:
yeh I think this should be `Abstract..` right? so use the `abstractEntityName` if it is there.

Reply:
ah, yes, I assumed `className` *was* `Abstract...`. Shouln't it be?

Reply:
it's the className for the entity (so that would be the `Counter` for instance). maybe className is confusing. But using `abstractEntityName` now.